build.zig

```diff
@@ -359,9 +365,109 @@ fn build_tigerbeetle(
     },
 ) void {
+    const out_filename = if (options.target.result.os.tag == .windows)
+        "tigerbeetle.exe"
+    else
+        "tigerbeetle";
+
+    const tigerbeetle_bin = if (options.multiversion) |version| bin: {
+        assert(!options.emit_llvm_ir);
+        const build_multiversion_exe = b.addExecutable(.{
+            .name = "build_multiversion",
+            .root_source_file = b.path("src/build_multiversion.zig"),
+            // Enable aes extensions for vsr.checksum on the host.
+            .target = resolve_target(b, null) catch @panic("unsupported host"),
+        });
+        // Ideally, we should pass `vsr_options` here at runtime. Making them comptime
+        // parameters is inelegant, but practical!
+        build_multiversion_exe.root_module.addOptions("vsr_options", options.vsr_options);
+
+        const build_multiversion = b.addRunArtifact(build_multiversion_exe);
+        if (options.target.result.os.tag == .macos) {
+            build_multiversion.addArg("--target=macos");
+            inline for (.{ "x86_64", "aarch64" }, .{ "x86-64", "aarch64" }) |arch, flag| {
+                build_multiversion.addPrefixedFileArg(
+                    "--tigerbeetle-current-" ++ flag ++ "=",
+                    build_tigerbeetle_executable(b, .{
+                        .vsr_module = options.vsr_module,
+                        .vsr_options = options.vsr_options,
+                        .target = resolve_target(b, arch ++ "-macos") catch unreachable,
+                        .mode = options.mode,
+                        .tracer_backend = options.tracer_backend,
+                    }).getEmittedBin(),
+                );
+            }
+        } else {
+            build_multiversion.addArg(b.fmt("--target={s}-{s}", .{
+                @tagName(options.target.result.cpu.arch),
+                @tagName(options.target.result.os.tag),
+            }));
+            build_multiversion.addPrefixedFileArg(
+                "--tigerbeetle-current=",
+                build_tigerbeetle_executable(b, .{
+                    .vsr_module = options.vsr_module,
+                    .vsr_options = options.vsr_options,
+                    .target = options.target,
+                    .mode = options.mode,
+                    .tracer_backend = options.tracer_backend,
+                }).getEmittedBin(),
+            );
+        }
+
+        if (options.mode == .Debug) {
+            build_multiversion.addArg("--debug");
+        }
+
+        build_multiversion.addPrefixedFileArg(
+            "--tigerbeetle-past=",
+            download_release(b, version, options.target, options.mode),
+        );
+        build_multiversion.addArg(b.fmt(
+            "--tmp={s}",
+            .{b.cache_root.join(b.allocator, &.{"tmp"}) catch @panic("OOM")},
+        ));
+        break :bin build_multiversion.addPrefixedOutputFileArg("--output=", out_filename);
+    } else bin: {
+        const tigerbeetle_exe = build_tigerbeetle_executable(b, .{
```

Comment:
Can we do the same thing at the first branch of this `if` and extract the logic into a function?

```zig
if (options.multiversion) |version| bin: {
  assert(!options.emit_llvm_ir);
  const build_multiversion_exe = build_tigerbeetle_multiversion_executable(b, .{ ... });
  break :bin build_multiversion_exe;
} else bin: {
  const tigerbeetle_exe = build_tigerbeetle_executable(b, .{ ... });
  // ...
  break :bin tigerbeetle_exe.getEmittedBin();
}
```

Reply:
Excellent call, I forgot that we don't do large functions anymore :rofl: